src/transformers/models/encoder_decoder/modeling_encoder_decoder.py

```diff
@@ -175,6 +175,12 @@ def __init__(
 
         self.encoder = encoder
         self.decoder = decoder
+        
+        # we can not stop the users from calling the encoder or decoder individually,
+        # we need to link the individual configs with the encoderdecoderconfig to ensure consistency
```

Comment:
```suggestion
        # make sure that the individual model's config refers to the shared config
        # so that the updates to the config will be synced
```